src/components/header/header.yaml

```diff
@@ -641,3 +649,66 @@ examples:
       tag:
         text: alpha
       html: This is a new service – your <a class="govuk-link" href="#">feedback</a> will help us to improve it.
+
+- name: with service navigation
+  data:
+    serviceNavigation:
+      serviceName: Service name
+      serviceUrl: '#s1'
+
+- name: with service navigation and govukHeader
+  data:
+    serviceName: Service name
+    serviceUrl: '#s1'
+    serviceNavigation:
+      serviceName: Service name
+      serviceUrl: '#s1'
+      classes: hmrc-service-navigation--with-language-select
+      slots:
+        end: | # example of possible rendered output of the hmrcServiceNavigationLanguageSelect component
+          <nav class="hmrc-service-navigation-language-select" aria-label="Language switcher">
+              <ul class="hmrc-service-navigation-language-select__list">
+                <li class="hmrc-service-navigation-language-select__list-item">
+                  <span aria-current="true">ENG</span>
+                </li>
+                <li class="hmrc-service-navigation-language-select__list-item">
+                  <a href="/components/service-navigation-language-select/welsh/preview" hreflang="cy" lang="cy" rel="alternate" class="govuk-link govuk-service-navigation__link" data-journey-click="link - click:lang-select:Cymraeg">
+                    CYM<span class="govuk-visually-hidden"> – Newid yr iaith ir Gymraeg</span>
+                  </a>
+                </li>
+              </ul>
+            </nav>
+    languageToggle:
+      en:
+        href: "/components/header/with-language-toggle-english/preview"
+      cy:
+        href: "/components/header/with-language-toggle-welsh/preview"
```

Comment:
@timsb I've updated the example here so it includes the class that's needed

I wish that we could do object destructuring in nunjucks to make it easier to wrap another component

because it feels like it would be so much nicer to have a hmrcServiceNavigation fork that had language select params in it and handled based on that the rendering of language select in the right place and adding of the class

but I know it will be like hmrcFooter where we have to keep track of all the params and pass them through https://github.com/hmrc/hmrc-frontend/blob/main/src/components/footer/template.njk#L18-L36 which makes it easy to miss new params being added during updates and just makes breakages easier during updates to govuk-frontend :(

Reply:
nothing in nhsuk-frontend, so guess it's more something from prototypes, if people find it challenging to use, then maybe it's just worth owning the extra maintenance for teams of a fork of service navigation with extra language select param and a bit of verbosity mapping params in it's template

a side effect of that would be that it would be quite easy for us to tweak the service navigation for all teams in the future if we needed to 🤷‍♂️

Reply:
Destructuring would be a nicer way of handling this. Thanks for looking into nhsuk-frontend. We'll keep an eye on this release then and see how people are doing.
It might be an issue that creeps up over time as new people join and weren't here for the initial release